src/core_plugins/kibana/public/dashboard/actions/view.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Action } from 'redux';
+import { createAction } from 'redux-actions';
+import { Filters, Query, TimeRange } from 'ui/embeddable';
+import { PanelId } from '../selectors';
+
+export enum ViewActionTypeKeys {
+  UPDATE_VIEW_MODE = 'UPDATE_VIEW_MODE',
+  SET_VISIBLE_CONTEXT_MENU_PANEL_ID = 'SET_VISIBLE_CONTEXT_MENU_PANEL_ID',
+  MAXIMIZE_PANEl = 'MAXIMIZE_PANEl',
+  MINIMIZE_PANEL = 'MINIMIZE_PANEL',
+  UPDATE_IS_FULL_SCREEN_MODE = 'UPDATE_IS_FULL_SCREEN_MODE',
+  UPDATE_USE_MARGINS = 'UPDATE_USE_MARGINS',
+  UPDATE_HIDE_PANEL_TITLES = 'UPDATE_HIDE_PANEL_TITLES',
+  UPDATE_TIME_RANGE = 'UPDATE_TIME_RANGE',
+  UPDATE_FILTERS = 'UPDATE_FILTERS',
+  UPDATE_QUERY = 'UPDATE_QUERY',
+}
+
+export interface UpdateViewModeAction extends Action {
```

Comment:
I guess I could do just:

```
export interface EmbeddableIsInitializedAction
  extends Action<EmbeddableIsInitializedActionPayload> {}
```

using redux-actions type, but then the type is just a generic string, not mapped to a specific string. Though the way I'm using it now, using a specific string isn't helping at all.  Not unless I did something like this:

```
   return embeddableError(
        embeddables,
        (action as EmbeddableErrorAction).payload
      );
```

would it then throw an error if the second parameter of that function didn't match.

Or I can use Redux Action type and create the Kibana Action like you suggested:

```
export interface KibanaAction<T, P> extends RAction {
  type: T;
  payload: P;
}
export interface EmbeddableIsInitializingAction
  extends KibanaAction<
      'EMBEDDABLE_IS_INITIALIZING',
      EmbeddableIsInitializingActionPayload
    > {}
```

Can't seem to do that using redux-actions type because then I get an error about type T not mapping to string.

Anyway, @azasypkin - do you prefer one of these methods?  I don't really feel strongly either way....  Not sure if anyone else on the team has played around with typescript and redux yet.

Reply:
I think I'll go with your first suggestion so I can preserve the specific string mapping to the payload.  Forgive me pretty much just repeating exactly what you suggested to me in my follow up comment, lol, I'm a little out of it today.